src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationUtilities.cs

```diff
@@ -34,6 +34,12 @@ internal static class SemanticClassificationUtilities
             IClassificationService classificationService,
             ClassificationTypeMap typeMap)
         {
+            // Let LSP handle semantic classification when running in the LSP editor.
+            if (spanToTag.SnapshotSpan.Snapshot.TextBuffer.IsInLspEditorContext())
+            {
+                return;
+            }
```

Comment:
i don't really understand this, and the comment doesn't explain 'why'.  Even if we're in LSP... we still have semantic classification.  so why is that semantic classification not going through our regular codepaths.

Reply:
also, i don't really like BG work calling into mutable FG state (like teh ITExtBuffer) to ask it questinos.  If it's 100% threadsafe... i guess its' ok.. it's just *super* smelly.

Reply:
@CyrusNajmabadi I'll update the comment.  But basically this is the common tagger entrypoint for semantic classifications.  When running via LSP, we don't want our tagger to produce semantic classification tags, we want the LSP client's tagger to do that job.
And this check can be changed to just retrieve the workspace service.  I'll do that.

Reply:
It's much rather you update the taggers, not this internal helper.